flow/project.py

```diff
@@ -700,42 +829,43 @@ def __init__(self, name, operations=None, operation_directives=None,
         else:
             self.operation_directives = operation_directives
 
-    def _set_entrypoint_item(self, entrypoint, directives, key, default, job):
+    def _set_entrypoint_item(self, entrypoint, directives, key, default, jobs):
         """Set a value (executable, path) for entrypoint in command.
 
         Order of priority is the operation directives specified and
         then the project specified value.
         """
         entrypoint[key] = directives.get(key, entrypoint.get(key, default))
         if callable(entrypoint[key]):
-            entrypoint[key] = entrypoint[key](job)
+            entrypoint[key] = entrypoint[key](*jobs)
 
-    def _determine_entrypoint(self, entrypoint, directives, job):
+    def _determine_entrypoint(self, entrypoint, directives, jobs):
         """Get the entrypoint for creating a _JobOperation.
 
         If path cannot be determined, then raise a RuntimeError since we do not
         know where to point to.
         """
         entrypoint = entrypoint.copy()
-        self._set_entrypoint_item(entrypoint, directives, 'executable', sys.executable, job)
+        self._set_entrypoint_item(entrypoint, directives, 'executable', sys.executable, jobs)
 
         # If a path is not provided, default to the path to the file where the
         # FlowProject (subclass) is defined.
-        default_path = inspect.getfile(job._project.__class__)
-        self._set_entrypoint_item(entrypoint, directives, 'path', default_path, job)
+        # We are assuming that all the jobs belong from the same project
+        default_path = inspect.getfile(jobs[0]._project.__class__)
+        self._set_entrypoint_item(entrypoint, directives, 'path', default_path, jobs)
         return "{} {}".format(entrypoint['executable'], entrypoint['path']).lstrip()
 
-    def _resolve_directives(self, name, defaults, job):
+    def _resolve_directives(self, name, defaults, jobs):
         if name in self.operation_directives:
             directives = deepcopy(self.operation_directives[name])
         else:
             directives = deepcopy(defaults.get(name, dict()))
         nranks = directives.get('nranks', 1)
         nthreads = directives.get('omp_num_threads', 1)
         if callable(nranks) or callable(nthreads):
-            def np_callable(job):
-                nr = nranks(job) if callable(nranks) else nranks
-                nt = nthreads(job) if callable(nthreads) else nthreads
+            def np_callable(*jobs):
+                nr = nranks(*jobs) if callable(nranks) else nranks
+                nt = nthreads(*jobs) if callable(nthreads) else nthreads
                 return nr*nt
```

Comment:
`nranks` as defined [here](https://github.com/glotzerlab/signac-flow/pull/283/files#diff-dd7774bb5ebcb1df3843a90e78c026a7R257) is _The number of MPI ranks to use for this operation._
I thought that it'd always return an Integer (Given any number of jobs we pass if it's a callable)

Reply:
Sorry if this was unclear -- my actual question is "What does `nranks(job1, job2, job3)` return?" Does it return a list of integers (ranks for each job)? If it returns a list, then `nr` is a list. If you have `nr = [4, 2, 8]` and `nt = [1, 1, 1]`, then you have a problem because lists can't be multiplied (NumPy arrays _can_ be multiplied but I don't want to use NumPy here).

Reply:
I don't think a list of MPI ranks should be returned here, it should only return an integer. 
@b-butler please suggest.

Reply:
I think it should be an integer.

Reply:
I concur. We should return an integer. An aggregate operation is a singular operation that has a singular resource requirement (partitioning the resources internally is the user's responsibility).

Reply:
Clarified via Slack, marking as resolved. I was misunderstanding this code. Thanks @csadorf @b-butler.